lib/matplotlib/cbook.py

```diff
@@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
     Minor bugfixes by Michael Droettboom
     '''
     def __init__(self, cb):
+        self._hash = hash(cb)
+        self._callbacks = []
```

Comment:
Right, that's what I meant. I wasn't being very clear. Thank you.

Double-calling might seem innocuous at first, but it could very well be
that the callback might perform a non-repeatable action such as deleting an
artist, or maybe the second call would result in accessing the wrong item
in a list. This is a serious issue, and can be very difficult to debug.

On Thu, Feb 19, 2015 at 10:42 AM, OceanWolf <notifications@github.com>
wrote:

> In lib/matplotlib/cbook.py
> <https://github.com/matplotlib/matplotlib/pull/4118#discussion_r24996653>:
>
> > @@ -361,9 +361,11 @@ class _BoundMethodProxy(object):
> >      Minor bugfixes by Michael Droettboom
> >      '''
> >      def __init__(self, cb):
> > +        self._hash = hash(cb)
> > +        self._callbacks = []
>
> Well, they do get uniquely identified, the problem lies in that they have
> too many unique identifiers, they should have just one, each (signal,
> callback) pair should have a single unique id. This does not happen.
>
> As a knock on effect, if you were to trigger the signal in the test cases,
> the callback would get called twice, when it should only get called the
> once.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4118/files#r24996653>.
>